float32 position
---
bool state